src/day-four/script-one.ts

```diff
@@ -27,6 +27,25 @@ for (let password = rawInput[0]; password <= rawInput[1]; password++) {
 }
 
 console.log('the number of matching passwords is: ', matchingPasswords.length);
+function meetsCriteria(pw: number, range: number[]): boolean {
+	if (!hasValidLength(pw)) {
+		return false;
+	}
+	if (!isInRange(pw, range)) {
+		return false;
+	}
+	if (!isNeverDecreasing(pw)) {
+		return false;
+	}
+	if (!hasAdjacent(pw)) {
+		return false;
+	}
+	if (!adjacentHasNoLargerGroup(pw)) {
+		return false;
+	}
+	return true;
+}
+
 function hasValidLength(pw: number): boolean {
```

Comment:
@nicolasdular I have extracted all requirements into own functions.
You were right, this has made my life easier. Especially the last requirement got simpler because it allows me to include a simple:
`if (!hasAdjacent(pw)) {}`
as well.